### PR TITLE
Add externs for angular.$locale service

### PR DESCRIPTION
**Pull request by Joel-Kornbluh**
[Review on Reviewable](https://reviewable.io/reviews/google/closure-compiler/1497)

### CODE REVIEW

File: contrib/externs/angular-1.5.js [hunk 1]
```diff
@@ -2468,3 +2467,41 @@ angular.$timeout_.cancel = function(promise) {};
...
 /** @typedef {!Window} */
 angular.$window;
+
+/******************************************************************************
+ * $locale Service
+ *****************************************************************************/
+
+ /**
+ * @typedef {{
+ *   DATETIME_FORMATS: {
```
Comment (mprobst) [inline contrib/externs/angular-1.5.js]: According to the Angular docs, the only public API is the id field:
https://docs.angularjs.org/api/ng/service/$locale

Do you really need all those nested fields? They are supposed to be an implementation detail.

```js
/** @interface */
angular.$locale = function() {};
/** @type {string} */
angular.$locale.prototype.id;
```

Comment (ChadKillingsworth) [inline contrib/externs/angular-1.5.js]: I think a better question is do we want to encourage their use by including it in official externs?

Comment (mprobst) [inline contrib/externs/angular-1.5.js]: We don't :-)

**Comment by Dominator008** [comment]
Per the comment by @mprobst, these should not be included in the externs.

**Pull request closed without merging**